packages/frontend/src/components/VisualizationConfigs/ChartConfigPanel/CustomVisConfigTabs.tsx

```diff
@@ -65,6 +157,257 @@ const loadMonaco = (monaco: Monaco, schemas: Schema[]) => {
+                        label: field,
+                        kind: monaco.languages.CompletionItemKind.Class,
+                        insertText: field,
+                        range,
+                    };
+                },
+            );
+
+            return { suggestions };
+        },
+        triggerCharacters: ['$'],
+    });
+};
+const SelectTemplate = ({
+    itemsMap,
+    isCustomConfig,
+    // isEditorEmpty,
+    setEditorConfig,
+}: {
+    itemsMap: ItemsMap | undefined;
+    isCustomConfig: boolean;
+    isEditorEmpty: boolean;
+    setEditorConfig: (config: string) => void;
+}) => {
+    /*const [selectedTemplate, setSelectedTemplate] = useState<
+        TemplateType | undefined
+    >();*/
+
+    const loadTemplate = useCallback(
+        (template: TemplateType) => {
+            if (!isCustomConfig) return null;
+
+            /**
+             * When selecting a field for the x axis,
+             * we want to prioritize dimensions and date items
+             * over metrics and table calculations
+             */
+            const sortedItemsForX = Object.values(itemsMap || {}).sort(
+                (a, b) => {
+                    const getPriority = (item: ItemsMap[string]) => {
+                        if (isDimension(item) && isDateItem(item)) return 1;
+                        if (isDimension(item)) return 2;
+                        if (isCustomDimension(item)) return 3;
+                        if (isMetric(item)) return 4;
+                        return 5; // everything else
+                    };
+                    return getPriority(a) - getPriority(b);
+                },
+            );
+
+            /**
+             * When selecting a field for the y axis (and color/size values),
+             * we want to prioritize numeric metrics and table calculations
+             * over dimensions
+             */
+            const sortedItemsForY = Object.values(itemsMap || {}).sort(
+                (a, b) => {
+                    const getPriorityForY = (item: ItemsMap[string]) => {
+                        if (isMetric(item) && isNumericType(item.type))
+                            return 1;
+                        if (isMetric(item)) return 2;
+                        if (isTableCalculation(item)) return 3;
+                        return 4; // everything else
+                    };
+
+                    return getPriorityForY(a) - getPriorityForY(b);
+                },
+            );
+
+            const xField = sortedItemsForX[0];
+            const [yField, extraField] = sortedItemsForY;
+
+            const templateString = generateVegaTemplate(
+                template,
+                xField,
+                yField,
+                extraField,
+            );
+            setEditorConfig(templateString);
+        },
+        [isCustomConfig, itemsMap, setEditorConfig],
+    );
+    return (
+        <>
+            {/*
+            // confirmation modal ? yes or no
```

Comment:
I added this, but I think this is not needed, happy to remove it.

Reply:
I think this is not necessary and can be removed